### PR TITLE
Hide information when a student start a participation

### PR DESCRIPTION
**Pull request by rpadovani**
Goal of this PR is to make sure data about text exercise is never leak to students. Also, I fixed some issues reported on sentry.

I have implemented the following checks:
- when the user start a participation, do not provide any data about solutions
- restrict access to complaints
- restrict api call to example submissions
- restrict api call to example assessment

Fixed the following sentry errors:
- https://sentry.io/organizations/ls1intum/issues/996051141/?environment=test&project=1440029

### CODE REVIEW

**Comment by SiggZ** [comment]
I found some other places where we should hide sensitive information: https://github.com/ls1intum/ArTEMiS/pull/327

I merged your branch into mine, so maybe we could close this one?

File: src/main/java/de/tum/in/www1/artemis/web/rest/ExampleSubmissionResource.java [hunk 1]
```diff
@@ -100,7 +100,14 @@ public ExampleSubmissionResource(ExampleSubmissionService exampleSubmissionServi
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     public ResponseEntity<ExampleSubmission> getExampleSubmission(@PathVariable Long id) {
         log.debug("REST request to get ExampleSubmission : {}", id);
-        Optional<ExampleSubmission> exampleSubmission = exampleSubmissionService.get(id);
+        Optional<ExampleSubmission> exampleSubmission = exampleSubmissionService.getWithEagerExercise(id);
+
+        if (exampleSubmission.isPresent()) {
+            if (!authCheckService.isAtLeastInstructorForExercise(exampleSubmission.get().getExercise())) {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ExampleSubmissionResource.java]:
```suggestion
            if (! authCheckService.isAtLeastTeachingAssistantForExercise(exampleSubmission.get().getExercise())) {
```

File: src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java [hunk 2]
```diff
@@ -151,6 +151,17 @@ public ParticipationResource(ParticipationService participationService, CourseSe
         }
 
         Participation participation = participationService.startExercise(exercise, principal.getName());
+
+        // Hide information from students
+        if (!courseService.userHasAtLeastTAPermissions(course)) {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java]: please use participation.getExercise().filterSensitiveInformation() instead. This is much better and handles all exercises properly

**Comment by krusche** [comment]
Some changes have been included in another pull request. Please check the current version of develop

**Pull request merged**